src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/utils/GerritPluginChecker.java

```diff
@@ -0,0 +1,146 @@
+     * @return URL
+     */
+    private static String buildURL(IGerritHudsonTriggerConfig config) {
+        if (config == null) {
+            throw new IllegalArgumentException("Gerrit Server Configuration cannot be null");
+        }
+
+        if (!config.isUseRestApi()) {
+            logger.warn("REST API is not enabled.");
+            return null;
+        }
+
+        String gerritFrontEndUrl = config.getGerritFrontEndUrl();
+        String restUrl = gerritFrontEndUrl;
+        if (gerritFrontEndUrl != null && !gerritFrontEndUrl.endsWith("/")) {
+            restUrl = gerritFrontEndUrl + "/";
+        }
+        return restUrl;
+    }
+
+    /**
+     * Given a status code, decode its status.
+     * @param statusCode HTTP code
+     * @param pluginName plugin that was checked.
+     * @return true/false if installed or not.
+     */
+    private static boolean decodeStatus(int statusCode, String pluginName) {
+        switch (statusCode) {
+            case HttpURLConnection.HTTP_OK:
+                logger.info("Gerrit Plugin " + pluginName + " is installed");
+                return true;
+            case HttpURLConnection.HTTP_NOT_FOUND:
+                logger.info("Gerrit Plugin " + pluginName + " is NOT installed");
+                return false;
+            case HttpURLConnection.HTTP_UNAUTHORIZED:
+                logger.warn(NOT_ABLE_TO_VERIFY_IF_GERRIT_PLUGIN + pluginName
+                        + " is installed. User is not NOT AUTHORIZED! Error: " + Messages.HttpConnectionUnauthorized());
+                return false;
+            case HttpURLConnection.HTTP_FORBIDDEN:
+                logger.warn(NOT_ABLE_TO_VERIFY_IF_GERRIT_PLUGIN + pluginName
+                        + " is installed. User is Forbidden! Error: " + Messages.HttpConnectionUnauthorized());
+                return false;
+            default:
+                logger.warn(NOT_ABLE_TO_VERIFY_IF_GERRIT_PLUGIN + pluginName
+                        + " is installed. Error: " + Messages.HttpConnectionError(statusCode));
+                return false;
+        }
+    }
+    /**
+     * Query Gerrit to determine if plugin is enabled.
+     * @param config Gerrit Server Config
+     * @param pluginName The Gerrit Plugin name.
+     * @return true if enabled.
+     */
+    public static boolean isPluginEnabled(IGerritHudsonTriggerConfig config, String pluginName) {
+
+        String restUrl = buildURL(config);
+        if (restUrl == null) {
+            logger.warn(NOT_ABLE_TO_VERIFY_IF_GERRIT_PLUGIN + pluginName
+                    + " is installed. REST URL cannot be null. Is the REST API Enabled?");
+            return false;
+        }
+        logger.trace("{}plugins/{}/", restUrl, pluginName);
+
+        DefaultHttpClient httpclient = new DefaultHttpClient();
+        HttpGet httpGet = new HttpGet(restUrl + "plugins/" + pluginName + "/");
+        httpclient.getCredentialsProvider().setCredentials(new AuthScope(null, -1),
```

Comment:
You forgot to set the proxy.

I count now three places where a HttpClient is created in a similar manner with most likely cut & paste code.
A utility method in the config would help this a lot I think.